src/WebJobs.Script.WebHost/App_Start/WebHostResolver.cs

```diff
@@ -157,9 +157,12 @@ private static ScriptHostConfiguration CreateScriptHostConfiguration(WebHostSett
                 RootScriptPath = settings.ScriptPath,
                 RootLogPath = settings.LogPath,
                 FileLoggingMode = FileLoggingMode.DebugOnly,
-                TraceWriter = settings.TraceWriter
+                TraceWriter = settings.TraceWriter,
+                IsSelfHost = settings.IsSelfHost
```

Comment:
It is used by the host ID generation logic, so we need to make sure it is properly set before invoking that logic (just another one of the issues I'm trying to address with my other changes)